exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/compliant/CompliantTextRecordReader.java

```diff
@@ -101,16 +108,23 @@ public void setup(OperatorContext context, OutputMutator outputMutator) throws E
       InputStream stream = dfs.openPossiblyCompressedStream(split.getPath());
       TextInput input = new TextInput(settings,  stream, readBuffer, split.getStart(), split.getStart() + split.getLength());
 
-      TextOutput output = null;
       if(settings.isUseRepeatedVarChar()){
-        output = new RepeatedVarCharOutput(outputMutator, getColumns(), isStarQuery());
-      }else{
-        //TODO: Add field output.
-        throw new UnsupportedOperationException();
+        TextOutput output = new RepeatedVarCharOutput(outputMutator, getColumns(), isStarQuery());
+        this.reader = new TextReader(settings, input, output, whitespaceBuffer);
+        reader.start();
+      } else {
+        // two-phase read approach.
+        // phase-1: read the header from the file
+        String [] fieldNames = extractHeader(input);
+
+        // phase-2: now read the data
+        stream = dfs.openPossiblyCompressedStream(split.getPath());
+        input = new TextInput(settings,  stream, readBuffer, split.getStart(), split.getStart() + split.getLength());
+        TextOutput output = new FieldVarCharOutput(outputMutator, fieldNames, getColumns(), isStarQuery());
+        this.reader = new TextReader(settings, input, output, whitespaceBuffer);
```

Comment:
the second output is using FieldVarCharOutput
I made few cosmetic changes for better flow. will upload with other changes.